asserts/model_test.go

```diff
@@ -551,14 +565,21 @@ func (mods *modelSuite) TestClassicDecodeOK(c *C) {
 		},
 	})
 	// gadget included
-	reqSnaps := model.RequiredWithEssentialSnaps()
-	c.Check(reqSnaps, HasLen, len(allSnaps))
-	for i, r := range reqSnaps {
-		c.Check(r.SnapName(), Equals, allSnaps[i].Name)
-		c.Check(r.ID(), Equals, "")
+	reqSnaps := naming.NewSnapSet(model.RequiredWithEssentialSnaps())
+	for _, e := range essentialSnaps {
+		c.Check(reqSnaps.Contains(e), Equals, true)
+	}
+	for _, s := range snaps {
+		c.Check(reqSnaps.Contains(s), Equals, true)
 	}
 	// gadget excluded
-	c.Check(model.RequiredNoEssentialSnaps(), DeepEquals, reqSnaps[1:])
+	noEssential := naming.NewSnapSet(model.RequiredNoEssentialSnaps())
+	for _, e := range essentialSnaps {
+		c.Check(noEssential.Contains(e), Equals, false)
+	}
+	for _, s := range snaps {
+		c.Check(noEssential.Contains(s), Equals, true)
+	}
```

Comment:
Same about also checking that all reqSnaps / noEssential were actually verified.